tests/Fixtures/AlgoliaEngineTestModelCustomKey.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Tests\Fixtures;
+
+class AlgoliaEngineTestModelCustomKey extends TestModel
+{
+    public function getSearchableKey()
+    {
+        return "custom-{$this->getKey()}";
+    }
+
+    public function getReverseSearchableKey($key = '')
+    {
+        return str_replace('custom-1', '', $key);
```

Comment:
I don't think your tests are functioning how you expect. I can return literally any value from this method and the tests still pass.